parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -613,7 +613,26 @@ installKubeletKubectlAndKubeProxy() {
             if [[ ! -z ${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER} ]]; then
                 # network isolated cluster
                 echo "Detect Bootstrap profile artifact is Cache, will use oras to pull artifact binary"
-                registry_url="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER}/${K8S_REGISTRY_REPO}/kubernetes-node:v${KUBERNETES_VERSION}-linux-${CPU_ARCH}"
+                binary_version="v${KUBERNETES_VERSION}" # by default use Kubernetes versions
+                # however, the kubelet and kubectl binary version may different with kubernetes version due to hotfix or beta
+                # so that we still need to extract the binary version from kube_binary_url
+                url_regax='https://acs-mirror.azureedge.net/kubernetes/v[0-9]+\.[0-9]+\..+/binaries/.+'
```

Comment:
is the `https://acs-mirror.azureedge.net` bit different in sov/AGC? if so we'd also need to account for that